security/pkg/server/ca/server.go

```diff
@@ -88,29 +87,24 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 		ForCA:      false,
 		CertSigner: certSigner,
 	}
-	cert, signErr := s.ca.Sign([]byte(request.Csr), certOpts)
+	var signErr error
+	var cert []byte
+	var respCertChain []string
+	if certSigner == "" {
+		cert, signErr = s.ca.Sign([]byte(request.Csr), certOpts)
+	} else {
+		respCertChain, signErr = s.ca.SignWithCertChain([]byte(request.Csr), certOpts)
+	}
 	if signErr != nil {
 		serverCaLog.Errorf("CSR signing error (%v)", signErr.Error())
 		s.monitoring.GetCertSignError(signErr.(*caerror.Error).ErrorType()).Increment()
 		return nil, status.Errorf(signErr.(*caerror.Error).HTTPErrorCode(), "CSR signing error (%v)", signErr.(*caerror.Error))
 	}
-	respCertChain := []string{string(cert)}
-	if len(certChainBytes) != 0 {
-		respCertChain = append(respCertChain, string(certChainBytes))
-	}
-	if len(rootCertBytes) == 0 {
```

Comment:
this is for the multi-signer case only, so it is safe we remove the monitoring part as it is newly added as well.